### DISCUSSION TITLE
How to completely disable auto-imports in nuxt3?

### DISCUSSION THREAD

**Discussion by nikolay-zoteev** [nuxt/nuxt#16320] [Questions] [answered]
```
export default defineNuxtConfig({
	autoImports: {
		sources: [],
		global: false,
		dirs: [],
	},
	components: false,
	...
```
It didn't work for me

**Comment by nikolay-zoteev** [comment]
It's work
```
export default defineNuxtConfig({
	hooks: {
		'autoImports:sources': (sources) => sources.splice(0, sources.length),
		'autoImports:dirs': (dirs) => dirs.splice(0, dirs.length),
		'components:dirs': (dirs) => dirs.splice(0, dirs.length),
	},
	...
```

**Reply by Mythie** [reply]
This works but long term I'd prefer to have the option to not have magic auto imports everywhere. I understand this might be better DX for some but I'd rather not debug the eventual magic gone wrong.

**Reply by skattabrain** [reply]
100,000%

**Comment by cawa-93** [comment]
I still need a reliable and documented way to turn off auto-import.

- Auto-import does not work for the server part
- Auto-import doesn't work for everyone. You still have business logic modules, helpers, utilities, services that you have to import on your own (actually IDE will do everything for you).
- Only top level files can be auto-imported.
- Auto-import of asynchronous components (such as those created by [`defineAsyncComponent`](https://vuejs.org/api/general.html#defineasynccomponent)) is not possible.

In addition
- Auto-import complicates IDE code analysis (jet brains have issues with deep import analysis)
- It takes twice as long to go from calling a component to declaring it

**Reply by TheAlexLichter** [reply]
Feel free to raise an issue about it then 😊

**Comment by max5432112345** [comment]
@manniL if auto import is disabled then nuxt modules not work (need to redirect import from #imports to #app)

>@pinia/nuxt/dist/runtime/plugin.vue3.mjs: [unimport] failed to find "defineNuxtPlugin" imported from "#imports"

new problem on

**Comment by alexbassy** [comment, marked as answer]
@nikolay-zoteev [according to the docs](https://v3.nuxtjs.org/guide/concepts/auto-imports/#disable-auto-imports), it is possible to disable this functionality. Should the thread be marked as answered? 🙂

**Comment by vedmant** [comment]
Looks like autoImports option was removed from latest v3 release, so how to disable this now?

**Reply by TheAlexLichter** [reply]
No, this is [still available](https://nuxt.com/docs/guide/concepts/auto-imports/#disable-auto-imports).

**Comment by ByScripts** [comment]
I've really hard time with auto import.

I'm trying to disable it completely, but then I can no longer import my component from the base layer.

What is the correct way to disable auto-import at all when using Nuxt layers?

**Comment by AdrianFahrbach** [comment]
I'm also having problems with this.
Disabling auto-imports works fine for stuff like `ref`, `useState` etc, but not with components.
I still get a `components.d.ts` file and no errors when a component is not imported.

**Reply by DouglasAR01** [reply]
Same. I've waited since the pre-release of Nuxt 3 to have a way to completely remove the auto-imports.
I don't think they will do it.

**Reply by TheAlexLichter** [reply]
You absolutely can disable them for everything. See e.g. [this video](https://www.youtube.com/watch?v=YHdDraNNGCk&lc=UgyxtyxFoTmTM42-8pp4AaABAg).
